python_modules/dagster/dagster/_core/telemetry.py

```diff
@@ -77,8 +93,24 @@
     "BUILDKITE",  # https://buildkite.com/docs/pipelines/environment-variables
 }
 
+P = ParamSpec("P")
+T = TypeVar("T")
```

Comment:
would be nice to have some documentation here, or aliases for the chains of callables. Even as someone who has dealt a lot with the telemetry code, finding it very difficult to figure out what I'm looking at.

Reply:
Yeah this is a bit dense-- the signatures of higher order decorators always are. I think it's considerably clearere when we use the style of our user-facing higher-order decorators, separating the `target_fn` and `metadata` args.